web/src/main/java/org/cbioportal/web/parameter/GeneFilter.java

```diff
@@ -0,0 +1,125 @@
+package org.cbioportal.web.parameter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.AssertTrue;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class GeneFilter implements Serializable {
+
+    private Set<String> molecularProfileIds;
+    private List<List<String>> geneQueries;
+
+    private final String GENE_QUERY_PATTERN = "^(\\w+)[\\s]*?(?:\\:(?:[\\s]*(?:(AMP)|(HOMDEL))\\b)+)?$";
+
+    public enum CNAType {
+        AMP("AMP", 2), GAIN("GAIN", 1), HOMDEL("HOMDEL", -2), HETLOSS("HETLOSS", -1);
```

Comment:
We don't have this logic defines at backend? It seems pretty important. Maybe just in the importer code?

Reply:
Maybe we should move this Enum to a separate file so it can be shared?

Reply:
importer uses numeric convention 1,-1, etc. There is one `CNA` in `CnaEvent` class which is sample specific and even it is not accessible in `web` module.

Reply:
I agree with @zhx828 here.   Unfortunately, we cannot make a dep on core as a dependency on web was recently introduced in the core.  Can we move this Enum into the model package so that it can be shared in the future (once we start to decompose the core module)?